### DISCUSSION TITLE
Automated Deployment Strategies?

### DISCUSSION THREAD

**Discussion by Fish1** [pocketbase/pocketbase#1876] [Q&A] [answered]
Does Pocketbase have any automated deployment strategies? A couple things I'm looking for...

1) Environment variables that will be read to initialize the first admin.

2) A way to load a schema without needing to copy and paste. Could the framework read the json schema? Or an environment variable telling pocketbase the file to load?

**Comment by ganigeorgiev** [comment, marked as answer]
For both you can use migrations - https://pocketbase.io/docs/migrations/.

Migrations can be written in Go and JS. There is no much documentation for the latter yet, but you can find some examples in https://github.com/pocketbase/pocketbase/discussions/1464 and https://github.com/pocketbase/pocketbase/discussions/1354.

Also note that the prebuilt executable comes with automigrations enabled by default, meaning that every collection change from the UI will autogenerate a migration file with it in a `pb_migrations` directory. You can commit this directory to your repo. All migrations are automatically applied on `serve` (_to change the default `pb_migrations` location, you can use `--migrationsDir` flag_).

**Reply by Fish1** [reply]
Thank you for the fast response :D I'll read up on migrations and see if I can get it working!

**Reply by Fish1** [reply]
This works perfect for automatically building the schema! Thanks for pointing this out to me.

I'm still not seeing how this would help with initializing the first admin using environment variable or something.

I can see how I could probably code this functionality by grabbing environment variables, easy enough. Just curious if pocketbase already had this build in or not.

**Reply by ganigeorgiev** [reply]
There is no functionality out of the box for creating an admin with just env variables (_I'm not sure if this is a good idea in general_), but you could create the first admin within a migration. You can find a JS example in https://github.com/pocketbase/pocketbase/discussions/1354.

Console command(s) for managing admins is in the [roadmap](https://github.com/orgs/pocketbase/projects/2) and will be added sometime in the future but for now is a low priority.

**Reply by Fish1** [reply]
Command line would be interesting because it could be handled by our dockerfile. But that example you provided does do what we need. Thanks!

The only reason I'm looking for this is because when deploying to something like railway.app you access your application with a URL. Just don't want the first person to type in the URL to be able to create the admin user.
